### ISSUE TITLE
Cannot connect via config flow

### ISSUE THREAD

**Issue by ArcticPowder** [peribeir/homeassistant-rademacher#83]
![image](https://github.com/peribeir/homeassistant-rademacher/assets/116690828/95f60b88-c786-4d99-9ce2-7045cc97b89f)

![image](https://github.com/peribeir/homeassistant-rademacher/assets/116690828/42a88453-511f-435e-b7d5-76d9ac3e9fc0)

The hpgateway is correctly set in my local network and it works fine when I'm using the HomePilot application. The Gateway IP is 192.168.178.137

I already tried to connect the gateway via LAN. Same issue.

Any ideas?

**Comment by peribeir** [comment]
Hi @ArcticPowder 
Is the gateway accessible through your browser?
If you can maybe it's a problem of home assistant not being able to reach the gateway.

What type of HA installation do you have? is it HA OS, HA Container or HA Core?
If it's HA OS you can try to login and do a ping command to the IP of the gateway and see if it works.
For container you can also connect to the container and do the same ping command.

**Comment by ArcticPowder** [comment]
> Hi @ArcticPowder 
> 
> Is the gateway accessible through your browser?
> 
> If you can maybe it's a problem of home assistant not being able to reach the gateway.
> 
> 
> 
> What type of HA installation do you have? is it HA OS, HA Container or HA Core?
> 
> If it's HA OS you can try to login and do a ping command to the IP of the gateway and see if it works.
> 
> For container you can also connect to the container and do the same ping command.

Hey @peribeir 

Thank you very much for your answer. 


Negative, I can not reach the Gateway via browser. Is it supposed to show something? 

Uh I think I use the OS. Sry I'm pretty new to that stuff. 

Home Assistant 2023.6.2
Supervisor 2023.06.2
Operating System 10.3
Frontend 20230608.0 - latest

I pinged the IP via SSH in Home Assistant - there was no packet loss or connection failure so far. 

![image](https://github.com/peribeir/homeassistant-rademacher/assets/116690828/b4b643c3-e1c3-4248-8b95-de21d1f82483)

**Comment by peribeir** [comment]
You should be able to access the gateway UI through the browser by going to http://192.168.178.137/ or http://hpgateway/

Anyway, can you try to do the following in your ssh window in home assistant:
`$ curl http://192.168.178.137/v4/devices`

It should return a JSON similar to:
`{"response":"get_visible_devices","devices":[{"description":"Your device description","deviceGroup":2,"did":1,"hasErrors":0,"iconSetInverted":0,"iconSet":{"k":"iconset8"},"messages":[],"name":"Living Room Blinds","properties":{"closingContact":3,"dawn":3,"dusk":3,"motion":3,"rain":3,"smartphone":3,"smoke":3,"sun":3,"temperature":3,"time":3,"trigger":3,"warning":3,"wind":3},"statusValid":true,"statusesMap":{"Manuellbetrieb":0,"Position":72},"visible":true,"deviceNumber":"14234511","uid":"407903_1","voiceControlledBy":"Alexa,Google","origin":"HomePilot"}]}`

**Comment by ArcticPowder** [comment]
If i hit that in any kind of Browser i got this: 
![image](https://github.com/peribeir/homeassistant-rademacher/assets/116690828/e37d3900-8c1b-4121-905f-f06959aa9eef)

And if i type `curl http://192.168.178.137/v4/devices` into the SSH i get this:
![image](https://github.com/peribeir/homeassistant-rademacher/assets/116690828/83b94a90-af68-44c0-8215-6c159f4d4a29)

**Comment by peribeir** [comment]
That's very strange. All HomePilot Gateway I've come across should respond to this request.
It looks like either you have a different kind of gateway I'm not aware of (maybe an older model not compatible with this integration), or something is not well configured in your gateway.
Maybe a firmware update is needed (?)
What gateway model do you have? Is it HomePilot or the smaller Start 2 Smart gateway?
HomePilot: https://www.amazon.co.uk/Rademacher-HomePilot-3rd-Generation-Control/dp/B08MV1HMRZ?th=1
Start 2 Smart: https://www.amazon.co.uk/RADEMACHER-34000019-Bridge-Smart-Gateway/dp/B08JLXWV6K

**Comment by ArcticPowder** [comment]
Yeah I think so too. But I think I know what's causing that problem. I've got this one here: https://de.homepilot-smarthome.com/produkte/pakete/starterset-rollladen-smart#tabs-f99c737d-1000-4da5-a082-e11949d80ffa

Rademacher is now HomePilot. But it should be Hardware so far. The gateway should be very similar to the 3rd Generation of Rademacher's HomePilot. 

Nevertheless I should have a connection to the gateway via Browser or am I wrong? Or is there a special Port I have to connect to?

Best regards and thank you so far. I appreciate your help!

**Comment by ArcticPowder** [comment]
So, I have to come to terms with the fact that for my gateway is not compatible to HomeAssistant yet. 🥲

**Comment by peribeir** [comment]
> Yeah I think so too. But I think I know what's causing that problem. I've got this one here: https://de.homepilot-smarthome.com/produkte/pakete/starterset-rollladen-smart#tabs-f99c737d-1000-4da5-a082-e11949d80ffa
> 
> Rademacher is now HomePilot.
> 
> Nevertheless I should have a connection to the gateway via Browser or am I wrong? Or is there a special Port I have to connect to?
> 
> Here is the answer from the Customer Support: ”The Rademacher Smarthome Box (formerly Homepilot G3) unfortunately has no technical connection to the Gateway premium. The Gateway premium is solely app-controlled and is also not a successor or anything like that. It is an independent smarthome system that is compatible with the DuoFern radio standard by Rademacher as an ecosystem.“
> 
> Best regards and thank you so far. I appreciate your help!

I'm sure there is some kind of API for the new Gateway, but surely it is not compatible with the Rademacher HomePilot.
Unfortunately, since I don't own the new one, it will be impossible for me to provide the support you'll need in the integration.

The fact that you get a "404 Not found" response definitely means that there is an embedded webserver running. But it is difficult to reverse engineer the API if there is no webpage (it was the main reason it was easy to integrate the old Rademacher bridge).

If you ever find what is the API, feel free to open an issue, and maybe we can work together to expand this or even create a new integration for HomeAssistant.

For now, I'll close this Issue.
